src/unused_finder/unused_finder.rs

```diff
@@ -164,57 +149,81 @@ impl UnusedFinder {
-        self.refresh_graph(&flattened_walk_file_data);
-    }
-
-    // Given a Vec<WalkFileMetadata> refreshes the in-memory representation of imports/exports of source file graph
-    fn refresh_graph(&mut self, flattened_walk_file_data: &Vec<WalkFileMetaData>) {
-        let files: HashMap<String, Arc<GraphFile>> = flattened_walk_file_data
-            .par_iter()
-            .map(|file| {
-                (
-                    file.source_file_path.to_string(),
-                    Arc::new(GraphFile::new(
-                        file.source_file_path.clone(),
-                        file.import_export_info
-                            .exported_ids
-                            .iter()
-                            .map(|e| e.metadata.export_kind.clone())
-                            .collect(),
-                        file.import_export_info.clone(),
-                        self.entry_packages.clone().contains(&file.package_name), // mark files from entry_packages as used
-                    )),
-                )
-            })
-            .collect();
+        self.src_files = source_files;
 
-        self.graph = Graph { files };
+        self.test_files = test_files;
     }
 
-    #[napi]
     pub fn find_unused_items(
         &mut self,
         files_to_check: Vec<String>,
     ) -> napi::Result<UnusedFinderReport> {
         self.logs = vec![];
         self.logs.push(format!("{:?}", &files_to_check));
+        // Create graph containing only src_files
+        let mut graph = create_graph(&self.src_files, &self.entry_packages);
         if files_to_check
             .iter()
             .any(|file| !self.file_path_exported_items_map.contains_key(file))
         {
             self.logs.push("Refreshing all files".to_string());
-            self.refresh_file_list();
+            self.refresh_all_files();
         } else {
             self.logs.push("Refreshing only some files!".to_string());
-            self.refresh_files_to_check(&files_to_check);
+            self.refresh_files_to_check(&files_to_check, &mut graph);
         }
         // Clone file_path_exported_items_map to avoid borrow checker issues with mutable/immutable references of `self`.
-        let mut file_path_exported_items_map = self.file_path_exported_items_map.clone();
+        let file_path_exported_items_map = self.file_path_exported_items_map.clone();
+        // let entry_packages = self.entry_packages;
+        // Create binding to entry_packages to avoid borrow checker complain about borrowing `self`
+        let entry_packages = &self.entry_packages;
+        let entry_pkgs_files: Vec<String> = self
+            .src_files
+            .par_iter_mut()
+            .filter_map(|file| {
+                if entry_packages.contains(&file.package_name) {
+                    return Some(file.source_file_path.clone());
+                }
+                None
+            })
+            .collect();
```

Comment:
Rather than repeating this process twice, we should just include all test files in the entrypoint here.

Reply:
You didn't address this change, there's still two redundant frontier walks.